dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/filter/ConsumerContextFilter.java

```diff
@@ -54,7 +54,14 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
             return invoker.invoke(invocation);
         } finally {
             // TODO removeContext? but we need to save future for RpcContext.getFuture() API. If clear attachments here, attachments will not available when postProcessResult is invoked.
+            String tag = null;
+            if (RpcContext.getContext().getAttachments() != null && RpcContext.getContext().getAttachment(Constants.TAG_KEY) != null) {
+                tag = RpcContext.getContext().getAttachment(Constants.TAG_KEY);
+            }
             RpcContext.getContext().clearAttachments();
+            if (tag != null) {
+                RpcContext.getContext().setAttachment(Constants.TAG_KEY, tag);

```

Comment:
I don't think this default behavior should be changed.  We should clear the attachments each time to avoid effects on the following RPC calls.